### PR TITLE
new readme updates

### PR DESCRIPTION
**Pull request by mesameki**
Added intro, target audience, models...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,20 +157,29 @@ jupyter notebook
...
+* LIME Explainer: LIME Explainer uses the state-of-the-art Local interpretable model-agnostic explanations [(LIME)](https://github.com/marcotcr/lime) algorithm to create local surrogate models. Unlike the global surrogate models, LIME focuses on training local surrogate models to explain individual predictions.
+
+Tabular Explainer: Used with tabular datasets, it currently employs the following logic to invoke the Direct SHAP Explainers:
+* If it's a tree-based model, apply SHAP TreeExplainer, else
+* If it's a DNN model, apply SHAP DeepExplainer, else
+* If it's a linear model, apply SHAP LinearExplainer, else
+* Treat it as a black-box model and apply SHAP KernelExplainer
+ and  have docs
```
Comment (imatiach-msft) [inline README.md]: what is the "and have docs"?  is that a typo?

File: README.md [hunk 2]
```diff
@@ -23,6 +26,37 @@ This repository contains an SDK and Jupyter notebooks with examples to showcase
...
+Interpret-community is a practical and community-driven repository that hosts a wide range of community-developed machine learning interpretability techniques. This repository makes it easy for anyone involved in the development of an AI system to achieve transparency around their machine learning models. Data scientists, machine learning engineers, and researchers can easily add their own interpretability techniques via the set of extension hooks built into the parent repository, Interpret, and expand this repository to include their custom-made interpretability techniques.
+
+
+This repo incorporates community-driven state-of-the-art interpretability solutions under one roof with a unified set of data structures and API. Users could experiment with different interpretability methods, and/or add their custom-made interpretability techniques and perform comparative analysis to evaluate their brand new explainers. Using these tools, one can explain machine learning models globally on all data, or locally on a specific data point using the state-of-art technologies in an easy-to-use and scalable fashion. In particular, this released open source toolkit:
```
Comment (imatiach-msft) [inline README.md]: nit: maybe remove state-of-the-art at the start or end of the paragraph (it's mentioned twice in the same paragraph so seems redundant)

Comment (imatiach-msft) [reply README.md]: I see state-of-the-art mentioned 6 times in the doc, maybe we can use different wording for some of the cases

File: README.md [hunk 3]
```diff
@@ -23,6 +26,37 @@ This repository contains an SDK and Jupyter notebooks with examples to showcase
...
+2. Creates a common API across the integrated libraries    
+3. Applies optimizations to make it feasible to run on real-world datasets at scale   
+4. Provides improvements such as the capability to "reverse the feature engineering pipeline" to provide users with feature importance values in terms of raw features
+5. Provides interactive and exploratory visualization to empower data scientists to get the most level of insights about their data
+# <a name="comparison"></a> 
+
+# Interpret vs. Interpret-Community
+
+
+Interpret-Community and its parent, Interpret, both serve as a one-stop-shop tool for researchers, machine learning engineers, software developers, data scientists, and business executives to get insights about the model predictions and why model made the prediction it did. The parent repository, Interpret, hosts the interpretability techniques that are widely accepted by research community and have been published in prestigous  publications and press (e.g., SHAP, EBM). Interpret-Community is an extension to Interpret and hosts experimental interpretability techniques that are yet to receive acceptance from the research community.
```
Comment (imatiach-msft) [inline README.md]: and why model made the prediction it did -> and why the model made the predictions it did

Comment (imatiach-msft) [reply README.md]: by research community -> by the research community

Comment (imatiach-msft) [reply README.md]: prestigous  publications has extra double-space

Comment (imatiach-msft) [reply README.md]: hosts experimental interpretability techniques that are yet to receive acceptance from the research community and functionality for data scientists and software engineers to simplify their workflow

File: README.md [hunk 4]
```diff
@@ -123,20 +157,29 @@ jupyter notebook
...
+* [SHAP](https://github.com/slundberg/shap) Kernel Explainer: SHAP's Kernel explainer uses a specially weighted local linear regression to estimate SHAP values for any model.
+* [SHAP](https://github.com/slundberg/shap): SHAP's Linear Explainer computes SHAP values for a linear model, optionally accounting for inter-feature correlations.
+
+* Mimic Explainer: Mimic explainer is based on the idea of [global surrogate models](https://christophm.github.io/interpretable-ml-book/global.html)'s. A global surrogate model is an intrinsically interpretable model that is trained to approximate the predictions of a black box model as accurately as possible. Data scientist can interpret the surrogate model to draw conclusions about the black box model. This repo supporots the following interpretable models as surrogate model: LightGBM (LinearExplainableModel), Linear Regression (LinearExplainableModel), Stochastic Gradient Descent explainable model (SGDExplainableModel), and Decision Tree (DecisionTreeExplainableModel).
```
Comment (imatiach-msft) [inline README.md]: LightGBM (LinearExplainableModel) -> LightGBM (LGBMExplainableModel)

Comment (imatiach-msft) [reply README.md]: Linear Regression -> Linear/Logistic Regression

File: README.md [hunk 5]
```diff
@@ -23,6 +26,37 @@ This repository contains an SDK and Jupyter notebooks with examples to showcase
 - [Build Status](#BuildStatus)
 - [Additional References](#Refs)
 
+# <a name="intro"></a> 
+
+# Overview of Interpret-Community
```
Comment (bethz) [inline README.md]: in the contents, it's interpret-community and here it's Interpret-Community
good to pick one or the other

File: README.md [hunk 6]
```diff
@@ -23,6 +26,37 @@ This repository contains an SDK and Jupyter notebooks with examples to showcase
 - [Build Status](#BuildStatus)
 - [Additional References](#Refs)
 
+# <a name="intro"></a> 
+
+# Overview of Interpret-Community
+Interpret-community is a practical and community-driven repository that hosts a wide range of community-developed machine learning interpretability techniques. This repository makes it easy for anyone involved in the development of an AI system to achieve transparency around their machine learning models. Data scientists, machine learning engineers, and researchers can easily add their own interpretability techniques via the set of extension hooks built into the parent repository, Interpret, and expand this repository to include their custom-made interpretability techniques.
```
Comment (bethz) [inline README.md]: I'd suggest removing the - in community-driven and community-developed.
Separate note, the community is not driving so.....
Suggest change AI to ML, change achieve transparency to improve transparency

File: README.md (hunk 2 above)
Comment (bethz) [reply README.md]: suggest change solutions (we don't solve yet) to techniques or methods 
change perform to more easily perform
Suggest we pick either techniques or methods or tools as word so we are consistent.

File: README.md [hunk 7]
```diff
@@ -23,6 +26,37 @@ This repository contains an SDK and Jupyter notebooks with examples to showcase
...
+
+
+This repo incorporates community-driven state-of-the-art interpretability solutions under one roof with a unified set of data structures and API. Users could experiment with different interpretability methods, and/or add their custom-made interpretability techniques and perform comparative analysis to evaluate their brand new explainers. Using these tools, one can explain machine learning models globally on all data, or locally on a specific data point using the state-of-art technologies in an easy-to-use and scalable fashion. In particular, this released open source toolkit: 
+1. Actively incorporates state-of-the-art experimental interpretability techniques, and allows for further expansion by researchers and data scientists 
+2. Creates a common API across the integrated libraries    
+3. Applies optimizations to make it feasible to run on real-world datasets at scale
```
Comment (bethz) [inline README.md]: feasibl -> possible

File: README.md [hunk 8]
```diff
@@ -23,6 +26,37 @@ This repository contains an SDK and Jupyter notebooks with examples to showcase
...
+
+
+This repo incorporates community-driven state-of-the-art interpretability solutions under one roof with a unified set of data structures and API. Users could experiment with different interpretability methods, and/or add their custom-made interpretability techniques and perform comparative analysis to evaluate their brand new explainers. Using these tools, one can explain machine learning models globally on all data, or locally on a specific data point using the state-of-art technologies in an easy-to-use and scalable fashion. In particular, this released open source toolkit: 
+1. Actively incorporates state-of-the-art experimental interpretability techniques, and allows for further expansion by researchers and data scientists 
+2. Creates a common API across the integrated libraries    
+3. Applies optimizations to make it feasible to run on real-world datasets at scale   
+4. Provides improvements such as the capability to "reverse the feature engineering pipeline" to provide users with feature importance values in terms of raw features
```
Comment (bethz) [inline README.md]: add some clarity on:
to provide users with feature importance values in terms of raw features

File: README.md [hunk 9]
```diff
@@ -23,6 +26,37 @@ This repository contains an SDK and Jupyter notebooks with examples to showcase
...
+5. Provides interactive and exploratory visualization to empower data scientists to get the most level of insights about their data
```
Comment (bethz) [inline README.md]: get the most level of insights... -> gain significant insight into their data

File: README.md (hunk 3 above)
Comment (bethz) [inline README.md]: two git-hubs is actually a two-stop-shop.

Comment (bethz) [inline README.md]: and why model -> and insight into why the

Comment (bethz) [inline README.md]: Is Interpret intended to be a parent or a peer repo?

Comment (bethz) [inline README.md]: can we say Interpret is validated?

File: README.md [hunk 10]
```diff
@@ -23,6 +26,37 @@ This repository contains an SDK and Jupyter notebooks with examples to showcase
...
+
+Interpret-Community and its parent, Interpret, both serve as a one-stop-shop tool for researchers, machine learning engineers, software developers, data scientists, and business executives to get insights about the model predictions and why model made the prediction it did. The parent repository, Interpret, hosts the interpretability techniques that are widely accepted by research community and have been published in prestigous  publications and press (e.g., SHAP, EBM). Interpret-Community is an extension to Interpret and hosts experimental interpretability techniques that are yet to receive acceptance from the research community.            
+
+
+
+
+  # <a name="target"></a> 
+
+ # Target Audience
+1. Researchers: Interpret's extension hooks make it easy to extend and thus, researchers can easily add their interpretability solutions to the community repository and compare it to state-of-the-art interpretability techniques and other community solutions.
```
Comment (bethz) [inline README.md]: Agree with Researchers but what kind of Researchers? ML  Researchers interested in using an easy-to-use Interpretability  repo or Interpretability Researchers adding their own.

File: README.md [hunk 11]
```diff
@@ -23,6 +26,37 @@ This repository contains an SDK and Jupyter notebooks with examples to showcase
...
+
+2. Developers/Data Scientists: Having all of the state-of-the-art algorithms in one place makes it easy for data scientist to experiment with different interpretability techniques, and explain their model in a scalable and seamless manner. The set of rich interactive visualizations allow customers to train and deploy more transparent machine learning models instead of wasting time and effort on generating customized visualizations, addressing scalability issues by optimizing third-party interpretability techniques, and adopting/operationalizing interpretability solutions.
```
Comment (bethz) [inline README.md]: Suggest splitting developers and data sci into two bullets.

File: README.md [hunk 12]
```diff
@@ -23,6 +26,37 @@ This repository contains an SDK and Jupyter notebooks with examples to showcase
...
+
+
+
+  # <a name="target"></a> 
+
+ # Target Audience
+1. Researchers: Interpret's extension hooks make it easy to extend and thus, researchers can easily add their interpretability solutions to the community repository and compare it to state-of-the-art interpretability techniques and other community solutions.
+
+2. Developers/Data Scientists: Having all of the state-of-the-art algorithms in one place makes it easy for data scientist to experiment with different interpretability techniques, and explain their model in a scalable and seamless manner. The set of rich interactive visualizations allow customers to train and deploy more transparent machine learning models instead of wasting time and effort on generating customized visualizations, addressing scalability issues by optimizing third-party interpretability techniques, and adopting/operationalizing interpretability solutions. 
+3. Business Executives: The core logic and visualizations are beneficial for raising awareness among those involved in developing AI applications, allow them to audit the output of models used by AI applications, and establish a strong governance framework around the use of AI applications.
```
Comment (bethz) [inline README.md]: the output of models used by AI apps-> audit model predictions for potential bias and use this insight to help establish stronger governance capabilities.

File: README.md [hunk 13]
```diff
@@ -123,20 +157,29 @@ jupyter notebook
...
-* Global feature importances summarize the most importance features at the model-level
- * The KernelExplainer and MimicExplainer are for BlackBox models
- * The MimicExplainer is faster but less accurate than the KernelExplainer
- * The TreeExplainer is for tree-based models
- * The DeepExplainer is for DNN tensorflow or pytorch models
-[shap](https://github.com/slundberg/shap) and [lime](https://github.com/marcotcr/lime) have docs
+Any models that are trained on datasets in Python numpy.array, pandas.DataFrame, iml.datatypes.DenseData, or scipy.sparse.csr_matrix format are supported by this API.
+The explanation functions accept both models and pipelines as input. If a model is provided, the model must implement the prediction function predict or predict_proba that conforms to the Scikit convention. If a pipeline (name of the pipeline script) is provided, the explanation function assumes that the running pipeline script returns a prediction. The repository also supports models trained via PyTorch, TensorFlow, and Keras deep learning frameworks. The following are a list of the experimental explainers available in the community repository:
```
Comment (bethz) [inline README.md]: If a pipeline (name of pipeline script) -> If  a pipeline script is provided

File: README.md [hunk 14]
```diff
@@ -123,20 +157,29 @@ jupyter notebook
...
+
+* Mimic Explainer: Mimic explainer is based on the idea of [global surrogate models](https://christophm.github.io/interpretable-ml-book/global.html)'s. A global surrogate model is an intrinsically interpretable model that is trained to approximate the predictions of a black box model as accurately as possible. Data scientist can interpret the surrogate model to draw conclusions about the black box model. This repo supporots the following interpretable models as surrogate model: LightGBM (LinearExplainableModel), Linear Regression (LinearExplainableModel), Stochastic Gradient Descent explainable model (SGDExplainableModel), and Decision Tree (DecisionTreeExplainableModel).
+* Permutation Feature Importance Explainer: Permutation Feature Importance is a technique used to explain classification and regression models that is inspired by [Breiman's Random Forests paper](https://www.stat.berkeley.edu/%7Ebreiman/randomforest2001.pdf) (section 10). At a high level, the way it works is by randomly shuffling data one feature at a time for the entire dataset and calculating how much the performance metric of interest decreases. The larger the change, the more important that feature is.
```
Comment (bethz) [inline README.md]: the way it works is-> it works by

File: README.md [hunk 15]
```diff
@@ -123,20 +157,29 @@ jupyter notebook
...
+* LIME Explainer: LIME Explainer uses the state-of-the-art Local interpretable model-agnostic explanations [(LIME)](https://github.com/marcotcr/lime) algorithm to create local surrogate models. Unlike the global surrogate models, LIME focuses on training local surrogate models to explain individual predictions.
+
+Tabular Explainer: Used with tabular datasets, it currently employs the following logic to invoke the Direct SHAP Explainers:
```
Comment (bethz) [inline README.md]: suggest to put the items below in a table. The "If it's a ..."

**Pull request merged**